src/android/NotificationHandler.java

```diff
@@ -0,0 +1,141 @@
+        float speed = 0;
+        int inProgress = 0;
+
+        for (Map.Entry<String, TaskData> entry : tasks.entrySet()) {
+            if (entry.getValue().getStatus() == TaskStatus.InProgress) {
+                inProgress++;
+
+                speed += convertUnitToKbps(
+                        entry.getValue().getInfo().getUploadRate().getUnit().name(),
+                        entry.getValue().getInfo().getUploadRate().getValue()
+                );
+            }
+        }
+
+        String pkg = mContext.getApplication().getPackageName();
+        String layoutDef = "layout";
+        String idDef = "id";
+
+        Resources resources = mContext.getResources();
+
+        RemoteViews notificationLayout = new RemoteViews(mContext.getPackageName(),
+                resources.getIdentifier("notification_small", layoutDef, pkg));
+
+        notificationLayout.setTextViewText(
+                resources.getIdentifier("notification_title", idDef, pkg),
+                inProgress == 0 ? defaultTitle : String.format("%s uploads remaining", uploadCount)
+        );
+
+        notificationLayout.setTextViewText(
+                resources.getIdentifier("notification_content_left", idDef, pkg),
+                inProgress == 0 ? defaultContent : String.format("%d in progress", inProgress)
+        );
+
+        notificationLayout.setTextViewText(
+                resources.getIdentifier("notification_content_right", idDef, pkg),
+                inProgress == 0 ? "" : toReadable(speed)
+        );
+
+        return builder
+                .setStyle(new NotificationCompat.DecoratedCustomViewStyle())
+                .setCustomContentView(notificationLayout)
+                .setSmallIcon(android.R.drawable.ic_menu_upload);
+    }
+
+    private float convertUnitToKbps(String unit, int speed) {
+        final String BPS = UploadRate.UploadRateUnit.BitPerSecond.name();
+        final String KPS = UploadRate.UploadRateUnit.KilobitPerSecond.name();
+        final String MPS = UploadRate.UploadRateUnit.MegabitPerSecond.name();
+
+        float value = 0;
+
+        if (unit == BPS) {
+            value = speed / 1000;
+        }
+
+        if (unit == KPS) {
+            value = speed;
+        }
+
+        if (unit == MPS) {
+            value = speed * 1000;
+        }
+
+        return value;
+    }
+
+    private String toReadable(float speed) {
+        final String KBPS = "Kbps";
+        final String MBPS = "Mbps";
+
+        String value = "";
+
+        if (speed >= 1000) {
+            value = String.format("%.2f %s", speed / 1000, MBPS);
+        }
+
+        if (speed >= 100 && speed < 1000) {
+            value = String.format("%.2f %s", speed, KBPS);
+        }
+
+        if (speed > 0 && speed < 1) {
+            value = String.format("%.2f %s", speed, KBPS);
+        }
+
+        if (speed == 0) {
+            return String.format("0 %s", KBPS);
+        }
+
+        return value;
```

Comment:
```suggestion
        if (speed >= 1000) return String.format("%.2f %s", speed / 1000, MBPS);
        String.format("%.2f %s", speed, KBPS);
```
On fait le plus simple pour le moment... + le cas de 1-99 n'était pas géré

Reply:
done